perceval/backends/core/github.py

```diff
@@ -618,11 +644,13 @@ def pulls(self, from_date=None):
 
                 pull_number = issue["number"]
                 path = urijoin(self.base_url, 'repos', self.owner, self.repository, "pulls", pull_number)
-
                 r = self.fetch(path)
                 pull = r.text
-
-                yield pull
+                pull_dict = json.loads(pull)
```

Comment:
@valeriocos, from what I understand (and from what I have implemented), pull request reviews are treated as a separate component (like comments, review_comments, commits, etc.) to form a structure as shown below (the last few lines of [this](https://api.github.com/repos/chaoss/grimoirelab-tutorial/pulls/12) pull request data): 
```
{
..

    },
      "comments": 0,
      "review_comments": 0,
      "maintainer_can_modify": false,
      "commits": 2,
      "reviews": <some number>
      "additions": 9,
      "deletions": 0,
   ..
..
}
```

Of course, this would require modifying the API data. But, atleast from what I can tell, each kind of "comment" (review comment, issue comment, etc) has its own count. 

What do you think?

Reply:
It makes sense, but the risk of modifying the API data is that if at some point the GitHub people decide to include the field `reviews` (as we are doing or for other purposes), we will have to align our logic to the new change in upstream. This is one of the reason to modify the upstream data as less as possible.

If you want, we could [contact](https://developer.github.com/forum/) GitHub and ask to include the attribute `reviews` to the data returned when fetching pull requests . I guess it should be a good feature for them since it will reduce the number of requests to their servers (a request will be done only if the reviews number is greater than zero). What do you think ?

Reply:
> This is one of the reason to modify the upstream data as less as possible.

Then where do you suggest including the reviews data?

> If you want, we could contact GitHub and ask to include the attribute reviews to the data returned when fetching pull requests

In my opinion, this is a good idea since reviews are slightly different from other kinds of comments mentioning them separately seems to make more sense. And as you mentioned, it will be more efficient.

Reply:
> Then where do you suggest including the reviews data?

just after `self.__init_extra_pull_fields(pull)`

I'm not sure if I'm missing something, if you want tomorrow I'll push the changes to a branch to simplify discussions ? what do you think ?

Reply:
> In my opinion, this is a good idea since reviews are slightly different from other kinds of comments mentioning them separately seems to make more sense. And as you mentioned, it will be more efficient.

Feel free to contact them or I can do it, as you prefer :)